frontend/src/views/new-project/NewProjectViewMachine.ts

```diff
@@ -0,0 +1,207 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { assign, Machine } from 'xstate';
+
+export interface NewProjectViewStateSchema {
+  states: {
+    newProjectView: {
+      states: {
+        pristine: {};
+        invalid: {};
+        valid: {};
+        submitted: {};
+        success: {};
+      };
+    };
+    toast: {
+      states: {
+        visible: {};
+        hidden: {};
+      };
+    };
+  };
+}
+
+export type SchemaValue = {
+  newProjectView: 'pristine' | 'invalid' | 'valid' | 'submitted' | 'success';
+  toast: 'visible' | 'hidden';
+};
+
+export interface NewProjectViewContext {
+  name: any;
+  message: string;
+  newProjectId: string;
+}
+
+export type ChangeNameEvent = { type: 'CHANGE_NAME'; name: string };
+export type HandleSuccessEvent = { type: 'HANDLE_SUCCESS'; id: string };
+export type HandleErrorEvent = { type: 'HANDLE_ERROR' };
+export type SubmitEvent = { type: 'SUBMIT' };
+export type ShowToastEvent = { type: 'SHOW_TOAST'; message: string };
+export type HideToastEvent = { type: 'HIDE_TOAST'; message: string };
+export type NewProjectEvent =
+  | ChangeNameEvent
+  | SubmitEvent
+  | HandleSuccessEvent
+  | HandleErrorEvent
+  | ShowToastEvent
+  | HideToastEvent;
+
+const isChangeNameEvent = (event: NewProjectEvent): event is ChangeNameEvent => {
+  return event.type === 'CHANGE_NAME';
+};
+const isHandleSuccessEvent = (event: NewProjectEvent): event is HandleSuccessEvent => {
+  return event.type === 'HANDLE_SUCCESS';
+};
+const isShowToastEvent = (event: NewProjectEvent): event is ShowToastEvent => {
+  return event.type === 'SHOW_TOAST';
```

Comment:
This is not needed since, in a way, it double checks something that XState will check for us. We talked about it on zoom but I'm still adding this comment for others to see.